pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/bestpractices/xml/AvoidReassigningLoopVariables.xml

```diff
@@ -577,7 +672,8 @@ public class Foo {
         ]]></code>
     </test-code>
 
-    <test-code>
+    <!-- Deactivated because this is hard to implement in the general case. -->
+    <test-code regressionTest="false">
```

Comment:
Should we create for that an issue to track it? It would be something that worked on pmd6 but in pmd7 we might produce false-positives/false-negatives.

"violation: various conditional reassignments of 'for' loop variable, skip allowed"

Reply:
Yes, this would be useful

Reply:
I created #3123 to track this, I think we could add other small differences in rule behavior, as documentation for pmd 7.